python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1259,6 +1300,57 @@ def _get_next_latest_storage_id(instance_queryer: CachingInstanceQueryer) -> int
     return max(next_latest_storage_id - cursor_offset, 0)
 
 
+def _asset_key_partition_key_iter_to_str(
+    asset_key_partition_keys: Iterable[AssetKeyPartitionKey],
+) -> str:
+    partition_keys_by_asset_key: Dict[AssetKey, List[str]] = {}
+    for asset_key_partition_key in asset_key_partition_keys:
+        partition_keys_by_asset_key.setdefault(asset_key_partition_key.asset_key, [])
+        if asset_key_partition_key.partition_key is not None:
+            partition_keys_by_asset_key[asset_key_partition_key.asset_key].append(
+                asset_key_partition_key.partition_key
+            )
+    return "\n".join(
+        f"- {asset_key.to_user_string()}: {{{', '.join(partition_keys)}}}"
+        if len(partition_keys) > 0
+        else f"- {asset_key.to_user_string()}"
+        for asset_key, partition_keys in partition_keys_by_asset_key.items()
+    )
+
+
+def _partition_subset_str(
+    partition_subset: PartitionsSubset,
+    partitions_def: PartitionsDefinition,
+    instance: DynamicPartitionsStore,
+):
+    return ",".join(
+        f"{partition_range.start} -> {partition_range.end}"
+        for partition_range in partition_subset.get_partition_key_ranges(
+            partitions_def=partitions_def, dynamic_partitions_store=instance
+        )
+    )
+
+
+def _asset_subsets_iter_to_str(
+    asset_graph_subset: AssetGraphSubset,
+    asset_graph: BaseAssetGraph,
+    instance: DynamicPartitionsStore,
+) -> str:
+    return_str = ""
+    asset_subsets = asset_graph_subset.iterate_asset_subsets(asset_graph)
+    for subset in asset_subsets:
+        if subset.is_partitioned:
+            partitions_def = asset_graph.get(subset.asset_key).partitions_def
+            partition_ranges_str = _partition_subset_str(
```

Comment:
Thoughts on only displaying ranges for time partitions defs?

I think for static partitions defs ranges don't make as much sense since partitions don't have a clear ordering.

Reply:
yeah good point